design/one-pager-stacks-security-isolation.md

```diff
@@ -241,13 +235,134 @@ permissionScope: Cluster
+
+With these two separate types, an administrator can allow access to installing cluster stacks to only certain restricted
+ roles, then perhaps be more permissive with access to installing namespaced stacks.
+
+If the install scope type does not match what the stack author has declared within their stack metadata, an error status
+ should be set on the install resource and installation of the stack should not proceed.
+
+### RBAC
+
+We will rely on RBAC to further define a clear separation of concerns for infrastructure resources, and workspace
+ resources in a namespace. In our model we assume that any Cluster Stack resources as they are global will be for the
+ infrastructure persona, whereas any namespace stack resources will be for the namespace that that stack is installed
+ into, thus the application persona.
+
+At install time:
+ - Cluster Stack: generate cluster roles for infra persona that auto-aggregate to built-in roles for
+ infrastructure owners.
+ - Namespaced Stack: generate cluster roles for the workspace persona different permission
 
-With these two separate types, an administrator can allow access to installing cluster stacks to only certain restricted roles, then perhaps be more permissive with access to installing namespaced stacks.
+#### Built in ClusterRoles
+
+The following built in roles will be installed and packaged with Crossplane, and updated with role aggregation as we
+ install additional stacks.
+
+The verbs supported in these roles(admin,edit,view) will model based on existing logic around
+[built-in kubernetes roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings).
+
+|ClusterRole Name | Permissions |
+|:-------|:----------|
+| crossplane:admin | Admin: environment, workspaces(namespaces), ClusterRole, Role, ClusterRoleBinding, RoleBinding |
+| crossplane:infra-admin | Admin: environment, manage cluster stacks |
+| crossplane:infra-edit  | Edit: environment, manage cluster stacks |
+| crossplane:infra-viewer | View: environment. Not permitted: secrets |
+
+If we create a workspace(specially annotated+labeled namespace), the stack-manager will create the corresponding cluster
+ roles for that workspace. This is for the benefit of external tools, that could role bind to this known clusterRole
+ that will be kept up to date automatically with ClusterRole Aggregation as the stack-manager installs and uninstalls
+ new stacks.
+
+ crossplane:ns-name:workspace-admin
+ crossplane:ns-name:workspace-edit
+ crossplane:ns-name:workspace-view
+
+To grant jane user admin access to the foo workspace, the following yaml would suffice
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+# This role binding allows "jane" to admin the Crossplane foo workspace.
+kind: RoleBinding
+metadata:
+  name: admin-workspace
+  namespace: foo
+subjects:
+- kind: User
+  name: jane # Name is case sensitive
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: ClusterRole
```

Comment:
interesting, so if the Stacks Manager created `ClusterRole` definitions for each Namespace Stack, and they were aggregated up to the well-known / build-in workspace-admin `ClusterRole`, then they a single `RoleBinding` could be used to grant access to all Namespace Stack CRDs in the cluster. Is that the behavior we're after? Sounds like we'd need a way to filter the CRDs that have been installed into a given namespace to help ensure there's an app stack controller listening in that namespace.

Reply:
Yes the admin concept is to make sure they can see everything. Due to our model it might make sense that crossplane admin only gets viewer rights  on crossplane namespaced stacks to avoid writing stuff in the wrong place.

Reply:
Yes the admin concept is to make sure they can see everything. Due to our model it might make sense that crossplane admin only gets viewer rights  on crossplane namespaced stacks to avoid writing stuff in the wrong place.

Reply:
fwiw, I wanted to verify that a clusterrole, restricted to a namespaced rolebinding, did restrict access to that kind from outside namespaces:
 
```
$ kubectl create clusterrole see-pods --verb=list --resource=pods
$ kubectl create rolebinding podseersight --clusterrole=see-pods --group=pod-seers
$ kubectl get pods -n crossplane-system --as-group=pod-seers --as=cluster-admin
Error from server (Forbidden): pods is forbidden: User "cluster-admin" cannot list resource "pods" in API group "" in the namespace "crossplane-system"
```

Looks good :+1: